app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,25 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: :index
+
+  def index
+    redirect_to new_user_session_path unless signed_in?
```

Comment:
here should be shown all posts, or it will be in next pr?

Reply:
in progress...